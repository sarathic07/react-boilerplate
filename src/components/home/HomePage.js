import React from "react";
import PropTypes from "prop-types";

const HomePage = props => {
  return (
    <div data-test={"home-comp"}>
      <div className={"greet"}>{"Welcome!"}</div>
      <input type={"button"} value={"Logout"} onClick={props.logout} />
    </div>
  );
};

HomePage.propTypes = {
  logout: PropTypes.func
};

export default HomePage;
