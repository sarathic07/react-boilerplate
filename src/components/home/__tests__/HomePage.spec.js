import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import HomePage from "../HomePage";
import { findByTestAtrr } from "./../../../../utils";

const setUp = (props = {}) => {
  const component = shallow(<HomePage {...props} />);
  return component;
};

describe("Home page", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render without errors", () => {
    const wrapper = findByTestAtrr(component, "home-comp");
    expect(wrapper.length).toBe(1);
  });

  it("match snapshot", () => {
    const wrapper = shallow(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should render without errors", () => {
    const wrapper = component.find(`.greet`);
    expect(wrapper.text()).toBe("Welcome!");
  });
});
