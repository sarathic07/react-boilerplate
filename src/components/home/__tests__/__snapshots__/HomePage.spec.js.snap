// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page match snapshot 1`] = `
<div
  data-test="home-comp"
>
  <div
    className="greet"
  >
    Welcome!
  </div>
  <input
    type="button"
    value="Logout"
  />
</div>
`;
